test/response_test.rb

```diff
@@ -852,9 +852,237 @@ class RubySamlTest < Minitest::Test
         signed_response = OneLogin::RubySaml::Response.new(document.to_s)
         settings.idp_cert = ruby_saml_cert_text
         signed_response.settings = settings
-        time = Time.parse("2015-03-18T04:50:24Z")
-        Time.stubs(:now).returns(time)
-        assert signed_response.is_valid?
+        Timecop.freeze(Time.parse("2015-03-18T04:50:24Z")) do
+          assert signed_response.is_valid?
+        end
+        assert_empty signed_response.errors
+      end
+    end
+
+    describe "retrieve nameID" do
+      it 'able when nameID inside the assertion' do
+        response_valid_signed.settings = settings
+        assert_equal "test@onelogin.com", response_valid_signed.name_id
+      end
+
+      it 'unable when encryptID inside the assertion but no private key' do
+          response_encrypted_nameid.settings = settings
+          assert_raises(OneLogin::RubySaml::ValidationError, "An EncryptedID found and no SP private key found on the settings to decrypt it") do
+            assert_equal "test@onelogin.com", response_encrypted_nameid.name_id    
```

Comment:
Right now here, I think we only have response.name_id     but I named responses that has a Encrypted NameID element:  response_encrypted_nameid

name_id is the response method to retrieve the NameID attribute from a SAML Response.
Remember that this method is here since long time ago:
https://github.com/onelogin/ruby-saml/blob/0.7.3/lib/onelogin/ruby-saml/response.rb#L37

I think that the right thing is to name it nameid, that is how I named it at the [php](https://github.com/onelogin/php-saml/blob/4c2bba447b9271b605d05c6c54eeabd7f7e64500/lib/Saml/Response.php#L31) and [python](https://github.com/onelogin/python-saml/blob/master/src/onelogin/saml2/response.py#L299).

The problem is that we can't change the name of this method since is a method that is already used in 3rd party apps.   Maybe create both methods for compatibility issues and start to use in our code the response.nameid  ??